src/main/java/sysc4806/project24/mini_survey_monkey/interceptors/SurveyStateInterceptor.java

```diff
@@ -0,0 +1,50 @@
+package sysc4806.project24.mini_survey_monkey.interceptors;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import sysc4806.project24.mini_survey_monkey.models.State;
+import sysc4806.project24.mini_survey_monkey.models.Survey;
+import sysc4806.project24.mini_survey_monkey.repositories.SurveyRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class SurveyStateInterceptor implements HandlerInterceptor {
+
+    private final SurveyRepository surveyRepository;
+    private List<State> stateToFilter = new ArrayList<>();
```

Comment:
**Change variables name FROM `stateToFilter` TO `statesToFilter`.**

Use pluralized nouns for lists.